control-plane-operator/controllers/hostedcontrolplane/ingress/router_test.go

```diff
@@ -50,3 +50,63 @@ func TestReconcileRouterServiceAnnotations(t *testing.T) {
 		t.Fatalf("expected target node labels annotation to be '%s', got %q", targetNodesLabel, got)
 	}
 }
+
+// Test that LoadBalancerSourceRanges is applied for external router services with allowedCIDRBlocks
+func TestReconcileRouterService_AppliesLoadBalancerSourceRanges(t *testing.T) {
+	// Given a HostedControlPlane on AWS
```

Comment:
If in the above conditional we only covers AWS, it's ok, but if not, should we add coverage for self-managed envs?